app/screens/AppNavigator/screens/Settings/components/RowLanguageItem.tsx

```diff
@@ -0,0 +1,73 @@
+import { LanguagePersistence } from '@api'
+import { ThemedIcon, ThemedText, ThemedTouchableOpacity } from '@components/themed'
+import { WalletAlert } from '@components/WalletAlert'
+import { useLanguageContext } from '@contexts/LanguageProvider'
+import { EnvironmentLanguage } from '@environment'
+import { NavigationProp, useNavigation } from '@react-navigation/native'
+import { tailwind } from '@tailwind'
+import { getLocaleByLanguageName, translate } from '@translations'
+
+import * as React from 'react'
+import { useCallback } from 'react'
+import { SettingsParamList } from '../SettingsNavigator'
+
+export function RowLanguageItem (props: { language: EnvironmentLanguage }): JSX.Element {
+  const navigation = useNavigation<NavigationProp<SettingsParamList>>()
+  const { language, setLanguage } = useLanguageContext()
+  const onPress = useCallback(async () => {
+    if (getLocaleByLanguageName(props.language) === language) {
+      return
+    }
+
+    WalletAlert({
+      title: translate('screens/Settings', 'Language Switch'),
+      message: translate(
+        'screens/Settings', 'You are about to change your language to {{language}}. Do you want to proceed?', { language: props.language }),
+      buttons: [
+        {
+          text: translate('screens/Settings', 'No'),
+          style: 'cancel'
+        },
+        {
+          text: translate('screens/Settings', 'Yes'),
+          style: 'destructive',
+          onPress: async () => {
+            setLanguage(getLocaleByLanguageName(props.language))
+            await LanguagePersistence.set(getLocaleByLanguageName(props.language))
+            navigation.goBack()
+          }
+        }
+      ]
+    })
+  }, [language])
+
+  const isLanguageSelected = (language: EnvironmentLanguage, languageStored: string): boolean => {
+    return getLocaleByLanguageName(language) === languageStored.slice(0, 2)
+  }
+
+  return (
+    <ThemedTouchableOpacity
+      onPress={onPress}
+      style={tailwind('flex flex-row p-4 pr-2 items-center justify-between')}
+      testID={`button_language_${props.language}`}
+    >
+      <ThemedText style={tailwind('font-medium')}>
+        {props.language}
```

Comment:
Language should be translated also. E.g, English is `Englisch` in German.

Reply:
@thedoublejay I was referring to the how iOS and FB handle the language selection screen, the main option is displayed in the target language, then the description is using the current chosen language.

Do you want us to follow them as well?

![fb_language](https://user-images.githubusercontent.com/40191153/132285319-5cc0d343-f444-4aec-ae07-d035e9f190ad.jpeg)
![ios_language](https://user-images.githubusercontent.com/40191153/132285326-fe1a9fde-7a4c-4e40-9217-928563b76b19.jpeg)

Reply:
Oh ok got it. You can check this with @jtlntno-cake. I'll let him decide which one should we use.

Just as a reference, on the desktop app we are translating every labels

Reply:
Yeah I like the suggestion. I incorporated that into the screen then.
Here's the [proto](https://www.figma.com/proto/SDIpZciuxJlUf4xf8yheiY/Light-Wallet?node-id=287%3A11410&viewport=241%2C48%2C0.57&scaling=scale-down&starting-point-node-id=841%3A28929&show-proto-sidebar=1)

Reply:
@jtlntno-cake from your proto every option is still in english tho, shouldn't the option itself use the language that it means? Like the screenshot i attached above

Reply:
Reviewed other applications - it makes sense to be aligned with the implementation from the screenshots above. Proto is now updated @kyleleow